lib/uefi/debug_support.cpp

```diff
@@ -59,3 +59,130 @@ EfiStatus efi_initialize_system_table_pointer(struct EfiSystemTable *system_tabl
 
   return SUCCESS;
 }
+
+static uint32_t efi_m_max_table_entries;
+
+#define EFI_DEBUG_TABLE_ENTRY_SIZE  (sizeof(union EfiDebugImageInfo))
+
+EfiStatus efi_core_new_debug_image_info_entry(uint32_t image_info_type,
+					      struct EFI_LOADED_IMAGE_PROTOCOL *loaded_image,
+					      EfiHandle image_handle) {
+  union EfiDebugImageInfo *table;
+  union EfiDebugImageInfo *new_table;
+  uint32_t index;
+  uint32_t table_size;
```

Comment:
this is not linux, we don't have to declare all variables on top of function. Declare them as you need them